src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1686,13 +1685,44 @@ class SetStatementSegment(BaseSegment):
                         ),
                     ),
                 ),
+                Sequence(
+                    Ref("ParameterNameSegment"),
+                    Ref("AssignmentOperatorSegment"),
+                    Ref("ExpressionSegment"),
+                ),
             ),
         ),
         Dedent,
         Ref("DelimiterGrammar", optional=True),
     )
 
 
+class AssignmentOperatorSegment(BaseSegment):
+    """One of the assignment operators.
+
+    Includes simpler equals but also +=, -=, etc.
+    """
+
+    match_grammar = OneOf(
```

Comment:
Classes must have a type:

```suggestion

    type = assignment_operator

    match_grammar = OneOf(
```